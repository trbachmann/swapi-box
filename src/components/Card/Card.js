import React from 'react';

export const Card = ({item, category}) => {
  let card;
  switch(item.type) {
    case 'people':
      card = <div>
        <h3>{item.name}</h3>
        <p>Homeworld: {item.homeworld_name}</p>
        <p>Species: {item.species_name}</p>
        <p>Population: {item.population}</p>
        <button>Add to Favorites</button>
      </div>;
      break;
    case 'planets':
      card = <div>
        <h3>Name</h3>
        <p>Terrain: </p>
        <p>Population: </p>
        <p>Climate:</p>
        <p>Residents: </p>
        <button>Add to Favorites</button>
      </div>;
      break;
    case 'vehicles':
      card = <div>
        <h3>{item.name}</h3>
        <p>Model: {item.model}</p>
        <p>Class: {item.vehicle_class}</p>
        <p>Number of Passengers: {item.passengers}</p>
        <button>Add to Favorites</button>
      </div>;
      break;
    default: 
      card = <h2>Sorry there are no cards for your selection. Please choose a different category.</h2>;
  }
  return(
    <div className='card'>
      {card}
    </div>
  )
}
