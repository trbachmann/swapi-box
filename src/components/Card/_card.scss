.card {
  border-radius: 15px;
  background: rgba(196, 196, 196, 0.31);
  height: 350px;

  .inner-card {
  margin: 1.2em;
  }
  
  .card-title {
    display: flex;
    justify-content: space-between;
    align-items: center;
  }

  h3 {
    color: $yellow;
  }

  p {
    color: white;
  }

  button {
    color: $yellow;
    font-size: 1em;
    margin: auto;
    border: solid $yellow 2px;
  }

  ul {
    color: white;
    height: 2.4em;
    overflow-y: scroll;
  }

  .favorite-false {
    color: white;
    margin-left: auto;
    &:hover {
      color: $yellow;
    }
  }

    .favorite-true {
    color: $yellow;
    margin-left: auto;
    &:hover {
      color: white;
    }
  }
}




