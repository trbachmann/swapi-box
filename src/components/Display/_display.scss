.card-display {
  display: grid;
  grid-template-columns: repeat(3, 250px);
  grid-gap: 2em;
  margin-bottom: 3em;
}