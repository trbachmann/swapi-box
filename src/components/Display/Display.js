import React from 'react';
import { Card } from '../Card/Card';
import PropTypes from 'prop-types';

export const Display = ({data, favorites, handleFavorite}) => {
    let cards = data.map(item => {
      let isFavorite;
      
      if (favorites.includes(item.created)){
        isFavorite = true;
      } else {
        isFavorite = false;
      }

      return <Card item={item} key={item.created} handleFavorite={handleFavorite} isFavorite={isFavorite}/>
    })
    
    return(
      <div className="card-display">
        {cards}
      </div>
    )
}

Card.propTypes = {
  item: PropTypes.object,
  handleFavorite: PropTypes.func,
  isFavorite: PropTypes.bool
}
