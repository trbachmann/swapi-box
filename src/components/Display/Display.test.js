import React from 'react';
import { Display } from './Display';
import { shallow } from 'enzyme';
import { Card } from '../Card/Card';

const mockPerson1 = { type: 'people', name: 'Han', homeworld_name: 'Alderaan', species_name: 'human', population: '200000', created: '239D' }
const mockPerson2 = { type: 'people', name: 'Luke', homeworld_name: 'Nabu', species_name: 'human', population: '502000', created: '239F'}
const mockFavorites = ['239F'];
const mockHandleFavorite = jest.fn();

describe('Display', () => {
  it('should match the snapshot with all data passed in', () => {
    const wrapper = shallow(
      <Display 
        data={[mockPerson1, mockPerson2]}
        favorites={mockFavorites}
        handleFavorite={mockHandleFavorite}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a card with the correct props', () => {
    const wrapper = shallow(<Display 
      data={[mockPerson1]}
      favorites={mockFavorites}
      handleFavorite={mockHandleFavorite}
    />)
    expect(wrapper.children(Card).length).toEqual(1);
    const { item, handleFavorite, isFavorite } = wrapper.find(Card).props();
    expect(item).toEqual(mockPerson1);
    expect(handleFavorite).toEqual(mockHandleFavorite);
    expect(isFavorite).toEqual(false);
  });
})