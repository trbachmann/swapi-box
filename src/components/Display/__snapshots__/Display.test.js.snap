// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display should match the snapshot with all data passed in 1`] = `
<div
  className="card-display"
>
  <Card
    handleFavorite={[MockFunction]}
    isFavorite={false}
    item={
      Object {
        "created": "239D",
        "homeworld_name": "Alderaan",
        "name": "Han",
        "population": "200000",
        "species_name": "human",
        "type": "people",
      }
    }
    key="239D"
  />
  <Card
    handleFavorite={[MockFunction]}
    isFavorite={true}
    item={
      Object {
        "created": "239F",
        "homeworld_name": "Nabu",
        "name": "Luke",
        "population": "502000",
        "species_name": "human",
        "type": "people",
      }
    }
    key="239F"
  />
</div>
`;
